src/test/unit/services/sample/standalone_gqs_bad_test.cpp

```diff
@@ -6,47 +6,44 @@
 #include <test/test-models/good/services/test_gq2.hpp>
 #include <test/unit/services/instrumented_callbacks.hpp>
 #include <test/unit/util.hpp>
+#include <Eigen/Dense>
 #include <gtest/gtest.h>
 #include <iostream>
+#include <string>
+#include <vector>
 
 class ServicesStandaloneGQ2 : public testing::Test {
 public:
-  ServicesStandaloneGQ2()
-    : model(context, &model_log) {}
+  ServicesStandaloneGQ2() :
+    logger(logger_ss, logger_ss, logger_ss, logger_ss, logger_ss) {}
+
+  void SetUp() {
+    stan::io::empty_var_context context;
+    model = new stan_model(context);
+  }
+  
+  void TearDown() {
+    delete(model);
+  }
 
-  stan::io::empty_var_context context;
-  std::stringstream model_log;
-  stan_model model;
-  std::stringstream sample_ss, logger_ss;
   stan::test::unit::instrumented_interrupt interrupt;
+  std::stringstream logger_ss;
+  stan::callbacks::stream_logger logger;
+  stan_model *model;
 };
 
 TEST_F(ServicesStandaloneGQ2, no_QoIs) {
-  stan::test::unit::instrumented_interrupt interrupt;
-  std::stringstream model_log;
-  std::stringstream sample_ss, logger_ss;
-  stan::callbacks::stream_writer sample_writer(sample_ss, "");
-  stan::callbacks::stream_logger logger(logger_ss,
-                                        logger_ss,
-                                        logger_ss,
-                                        logger_ss,
-                                        logger_ss);
+  std::vector<std::string> param_names;
+  std::vector<std::vector<size_t>> param_dimss;
+  stan::services::get_model_parameters(*model, param_names, param_dimss);
+  EXPECT_EQ(param_names.size(), 1);
+  EXPECT_EQ(param_dimss.size(), 1);
 
-  int num_params = stan::services::num_constrained_params(model);
-  EXPECT_EQ(num_params, 2);
-  std::vector<double> draw1;
-  draw1.push_back(-2.345);
-  draw1.push_back(-6.789);
-  std::vector<double> draw2;
-  draw2.push_back(-3.123);
-  draw2.push_back(-4.123);
-  std::vector<std::vector<double> > draws;
-  draws.push_back(draw1);
-  draws.push_back(draw2);
-  const std::vector<std::vector<double> > cdraws(draws);
-
-  int return_code = stan::services::standalone_generate(model,
-                                                        cdraws,
+  const Eigen::MatrixXd draws(1,1);
+  std::stringstream sample_ss;
+  stan::callbacks::stream_writer sample_writer(sample_ss, "");
+  int return_code = stan::services::standalone_generate(*model,
```

Comment:
Seems like this test should test that the test_gq2 model's print statement executes somehow, right?

Reply:
not relevant for this feature.
the only reason there's a print statement is to have a non-empty GQ block.

Reply:
What's this test testing?

Reply:
https://github.com/stan-dev/stan/blob/1812edf5274792006ed104388cc4a1d9a469af3d/src/stan/services/sample/standalone_gqs.hpp#L86

don't bother to run standalong generated quantities if no variables declared in the gq block.